client/tokenized-payment-request/blocks/index.js

```diff
@@ -20,6 +23,8 @@ const tokenizedCartPaymentRequestPaymentMethod = ( api ) => ( {
 	),
 	edit: <ApplePayPreview />,
 	canMakePayment: ( cartData ) => {
+		cartData = transformCartDataForStoreAPI( cartData, null );
+
```

Comment:
Should we maybe perform this transformation after these conditional checks beneath it? 

For example, if `getConfig( 'is_admin' )` is `true` and we are in the editor context, there likely will not be any `cartData` to manipulate. Similarly, if either `wcpayPaymentRequestParams` or `wcpayConfig` is `undefined`, I'd presume we'd not want to continue and simply return `false` immediately instead.

I'm basically proposing the below...

```js
	canMakePayment: ( cartData ) => {
		// If in the editor context, always return true to display the `edit` prop preview.
		// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4101.
		if ( getConfig( 'is_admin' ) ) {
			return true;
		}

		if ( typeof wcpayPaymentRequestParams === 'undefined' ) {
			return false;
		}

		if (
			typeof wcpayConfig !== 'undefined' &&
			wcpayConfig.isExpressCheckoutElementEnabled
		) {
			return false;
		}

		return api.loadStripe( true ).then( ( stripe ) => {
			// Create a payment request and check if we can make a payment to determine whether to
			// show the Payment Request Button or not. This is necessary because a browser might be
			// able to load the Stripe JS object, but not support Payment Requests.
			cartData = transformCartDataForStoreAPI( cartData, null );
			const pr = getPaymentRequest( {
				stripe,
				cartData,
			} );

			return pr.canMakePayment();
		} );
	},

```

Reply:
Yes, good call! Done in [37914dc](https://github.com/Automattic/woocommerce-payments/pull/9150/commits/37914dc4df9fe40985ca8f1704a7736263e3a4df)